lib/dry/component/container.rb

```diff
@@ -143,6 +143,14 @@ def self.require_component(key, &block)
           Kernel.require component.path
           yield(component.constant) if block
         else
+          imports.each do |ns, container|
+            begin
+              localized_key = key.sub(/^#{Regexp.escape(ns)}#{Regexp.escape(config.namespace_separator)}/, '')
+              return container.require_component(localized_key, &block)
+            rescue ArgumentError
+            end
+          end
+
```

Comment:
The only code/style question I have for this PR is whether you might like this `imports.each` block moved into a private method for succinctness/clarity of the main `.require_component` method. I thought about it briefly, but I thought there was a chance it might actually harm readability/understandability, since it obscures the early return that we need to get out of the method before that final `fail`.